examples/skybox/main.rs

```diff
@@ -0,0 +1,303 @@
+        let buf = device.create_buffer_mapped(vertex_data.len(), wgpu::BufferUsage::VERTEX).fill_from_slice(&vertex_data);
+
+        let stride = std::mem::size_of::<[f32; 2]>() as wgpu::BufferAddress;
+        let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+            bind_group_layouts: &[&bind_group_layout],
+        });
+
+        // Create the render pipeline
+        let pipeline = device.create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+            layout: &pipeline_layout,
+            vertex_stage: wgpu::ProgrammableStageDescriptor {
+                module: &vs_module,
+                entry_point: "main",
+            },
+            fragment_stage: Some(wgpu::ProgrammableStageDescriptor {
+                module: &fs_module,
+                entry_point: "main",
+            }),
+            rasterization_state: Some(wgpu::RasterizationStateDescriptor {
+                front_face: wgpu::FrontFace::Cw,
+                cull_mode: wgpu::CullMode::None,
+                depth_bias: 0,
+                depth_bias_slope_scale: 0.0,
+                depth_bias_clamp: 0.0,
+            }),
+            primitive_topology: wgpu::PrimitiveTopology::TriangleList,
+            color_states: &[wgpu::ColorStateDescriptor {
+                format: sc_desc.format,
+                color_blend: wgpu::BlendDescriptor::REPLACE,
+                alpha_blend: wgpu::BlendDescriptor::REPLACE,
+                write_mask: wgpu::ColorWrite::ALL,
+            }],
+            depth_stencil_state: None,
+            index_format: wgpu::IndexFormat::Uint16,
+            vertex_buffers: &[wgpu::VertexBufferDescriptor {
+                stride,
+                step_mode: wgpu::InputStepMode::Vertex,
+                attributes: &attributes,
+            }],
+            sample_count: 1,
+            sample_mask: !0,
+            alpha_to_coverage_enabled: false,
+        });
+
+        let sampler = device.create_sampler(&wgpu::SamplerDescriptor {
+            address_mode_u: wgpu::AddressMode::ClampToEdge,
+            address_mode_v: wgpu::AddressMode::ClampToEdge,
+            address_mode_w: wgpu::AddressMode::ClampToEdge,
+            mag_filter: wgpu::FilterMode::Nearest,
+            min_filter: wgpu::FilterMode::Linear,
+            mipmap_filter: wgpu::FilterMode::Nearest,
+            lod_min_clamp: -100.0,
+            lod_max_clamp: 100.0,
+            compare_function: wgpu::CompareFunction::Always,
+        });
+
+        let paths = [
+            "examples/skybox/images/posx.png",
+            "examples/skybox/images/posy.png",
+            "examples/skybox/images/posz.png",
+            "examples/skybox/images/negx.png",
+            "examples/skybox/images/negy.png",
+            "examples/skybox/images/negz.png",
+        ];
+	// we set these multiple times, but whatever
+	let (mut image_width, mut image_height) = (0, 0);
+        let faces = paths
```

Comment:
spacing is broken here (rogue tabs?)

Reply:
yes unfortunately it looks like the project hasn't adopted cargo fmt so i have to manually find these :(

Reply:
You are adding a new file, it costs you no friction to format it.

Reply:
> You are adding a new file, it costs you no friction to format it.

cargo fmt doesn't take files as arguments last i checked, unless something changed?

Reply:
rustfmt does